### PR TITLE
Avoid rust panic

### PR DESCRIPTION
**Pull request by warrenpuffett**
#1721

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,9 @@ env
 venv
 docs/_build/*
 target/
+typings/
+pyrightconfig.json
```
Comment (warrenpuffett) [inline .gitignore:6]: My editor use pyright to link/typecheck so need to ignore some config files related to it.

File: counterparty-rs/src/utils.rs [hunk 2]
```diff
@@ -40,10 +40,14 @@ fn script_to_address(script_pubkey: Vec<u8>, network: &str) -> PyResult<String>
             Err(_) => return Err(PyErr::new::<pyo3::exceptions::PyValueError, _>("Invalid version value")),
         };
 
+        let n = 22;
+        if script.len() < n{
```
Comment (ouziel-slama) [inline counterparty-rs/src/utils.rs:44]: hum.. should not be here 24 ? because we take script[2..22] ?

Comment (warrenpuffett) [reply counterparty-rs/src/utils.rs:44]: @ouziel-slama 22 is an index not the number of bytes to take: https://doc.rust-lang.org/std/ops/struct.Range.html. We're taking 20 bytes here, from the byte at index 2 up to the byte at index 22 (exclusive). We skipped 2 bytes so the length of the script must at least be 22.

Comment (ouziel-slama) [reply counterparty-rs/src/utils.rs:44]: ah right!! I forgot that you should never do a review before the second coffee lol!

**Pull request merged**